src/org/zaproxy/zap/extension/simpleExample/ExtensionSimpleExample.java

```diff
@@ -45,147 +46,151 @@
+
+    private static final String EXAMPLE_FILE = "example/ExampleFile.txt";
 
-	private static final String RESOURCE = "/org/zaproxy/zap/extension/simpleExample/resources";
-	
-	private static final ImageIcon ICON = new ImageIcon(
-			ExtensionSimpleExample.class.getResource( RESOURCE + "/cake.png"));
+    private ZapMenuItem menuExample;
+    private RightClickMsgMenu popupMsgMenuExample;
+    private AbstractPanel statusPanel;
 
-	private static final String EXAMPLE_FILE = "example/ExampleFile.txt";
-	
-	private ZapMenuItem menuExample;
-	private RightClickMsgMenu popupMsgMenuExample;
-	private AbstractPanel statusPanel;
+    private SimpleExampleAPI api = new SimpleExampleAPI(this);
 
     private static final Logger LOGGER = Logger.getLogger(ExtensionSimpleExample.class);
 
     public ExtensionSimpleExample() {
         super(NAME);
     }
-	
-	@Override
-	public void hook(ExtensionHook extensionHook) {
-	    super.hook(extensionHook);
-	    
-	    if (getView() != null) {
-	    	// Register our top menu item, as long as we're not running as a daemon
-	    	// Use one of the other methods to add to a different menu list
-	        extensionHook.getHookMenu().addToolsMenuItem(getMenuExample());
-	    	// Register our popup menu item
-	    	extensionHook.getHookMenu().addPopupMenuItem(getPopupMsgMenuExample());
-	    	// Register a 
-	    	extensionHook.getHookView().addStatusPanel(getStatusPanel());
-	    }
-
-	}
-
-	@Override
-	public boolean canUnload() {
-		// The extension can be dynamically unloaded, all resources used/added can be freed/removed from core.
-		return true;
-	}
-
-	@Override
-	public void unload() {
-		super.unload();
-
-		// In this example it's not necessary to override the method, as there's nothing to unload
-		// manually, the components added through the class ExtensionHook (in hook(ExtensionHook))
-		// are automatically removed by the base unload() method.
-		// If you use/add other components through other methods you might need to free/remove them
-		// here (if the extension declares that can be unloaded, see above method).
-	}
-
-	private AbstractPanel getStatusPanel() {
-		if (statusPanel == null) {
-			statusPanel = new AbstractPanel();
-			statusPanel.setLayout(new CardLayout());
-	        statusPanel.setName(Constant.messages.getString(PREFIX + ".panel.title"));
-	        statusPanel.setIcon(ICON);
-	        JTextPane pane = new JTextPane();
-			pane.setEditable(false);
-			// Obtain (and set) a font with the size defined in the options
-			pane.setFont(FontUtils.getFont("Dialog", Font.PLAIN));
-			pane.setContentType("text/html");
-			pane.setText(Constant.messages.getString(PREFIX + ".panel.msg"));
-			statusPanel.add(pane);
-		}
-		return statusPanel;
-	}
-
-	private ZapMenuItem getMenuExample() {
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        extensionHook.addApiImplementor(this.api);

```

Comment:
The minimum ZAP version should be updated (to at least 2.6.0) in the add-on manifest (ZapAddOn.xml).